signing_test.go

```diff
@@ -166,16 +172,39 @@ func TestSignerWithBrokenRand(t *testing.T) {
 
 	defer resetRandReader()
 
-	for _, alg := range sigAlgs {
+	for _, alg := range pssSigAlgs {
 		signingKey, verificationKey := GenerateSigningTestKey(alg)
-		for i, getReader := range readers {
+		for i, getReader := range brokenRandReaders {
 			RandReader = getReader()
 			err := RoundtripJWS(alg, serializer, corrupter, signingKey, verificationKey, "test_nonce")
 			if err == nil {
 				t.Error("signer should fail if rand is broken", alg, i)
 			}
 		}
 	}
+
+	// TODO(@pgporada) Remove this version checking
+	goVer, err := goVersionCheck()
+	if err != nil {
+		t.Error(err)
+	}
+	constraints, err := version.NewConstraint("< 1.20")
```

Comment:
I'd prefer not to import a whole new dependency just for this one test. The set of go versions that should trigger testing the RS algorithms is 1.12 through 1.19, so I'd just do a regex match against `"^1.1\d$"`.